kroxylicious/src/main/java/io/kroxylicious/proxy/config/PluginAnnotationIntrospector.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config;
+
+import java.lang.annotation.Annotation;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
+import com.fasterxml.jackson.databind.introspect.Annotated;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
+import com.fasterxml.jackson.databind.jsontype.TypeIdResolver;
+
+import io.kroxylicious.proxy.plugin.PluginConfig;
+
+public class PluginAnnotationIntrospector extends JacksonAnnotationIntrospector {
+
+    @Override
+    protected <A extends Annotation> A _findAnnotation(Annotated ann,
+                                                       Class<A> annoClass) {
+        if (annoClass == JsonTypeIdResolver.class) {
+            var pc = _findAnnotation(ann, PluginConfig.class);
+            if (pc != null) {
+                return (A) synthesizeJsonTypeIdResolver();
+            }
+        }
+        else if (annoClass == JsonTypeInfo.class) {
+            var pc = _findAnnotation(ann, PluginConfig.class);
+            if (pc != null) {
+                return (A) synthesizeJsonTypeInfo(pc);
+            }
+        }
+        return super._findAnnotation(ann, annoClass);
+    }
+
+    /**
+     * Returns a fake (i.e. not obtained through reflection)
+     * JsonTypeInfo instance which looks like this:
+     * <pre>{@code
+     * @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "type")
+     * }</pre>
+     * where "type" is taken from the {@link PluginConfig#instanceNameProperty()}.
```

Comment:
This explains the `what` but not the `why`. It would help me at least to document why we need this.